src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -73,6 +73,17 @@ const PersonalDetails = (() => {
                 get_settings.citizen;
         }
 
+        if (get_settings.tax_residence) {
+            get_settings.tax_residence =
+                get_settings.tax_residence
+                    .split(',')
+                    .reduce((all_residences, current_residence) =>
+                        `${all_residences + ((residence_list.find(obj => obj.value === current_residence) || {}).text || current_residence)  }, `, ''
+                    )
+                    .slice(0, -2) ||
```

Comment:
why not something simple like this?
```
get_settings.tax_residence
    .split(',')
    .map((current_residence) =>
        (residence_list.find(obj => obj.value === current_residence) || {}).text || current_residence
    ).join(', ')
```

Reply:
Because I hadn't thought of that approach